src/sysrepo.c

```diff
@@ -352,8 +352,17 @@ sr_connect(const sr_conn_options_t opts, sr_conn_ctx_t **conn_p)
         sr_conn_free(conn);
         if (created) {
             /* remove any created SHM so it is not considered properly created */
-            shm_unlink(SR_MAIN_SHM);
-            shm_unlink(SR_EXT_SHM);
+            char *shm_name = NULL;
+            if (!sr_path_main_shm(&shm_name)) {
```

Comment:
Actually, I did not realize that this is wrong because in the unlikely event that this fails, the `err_info` will leak. In cases when the error should not be propagated, you should just free it, it is printed when it is generated.

Reply:
right, I'll fix that

Reply:
Can I use sr_errinfo_merge ?

Reply:
Yes, I only looked just now that it is executed only when some error occurred so I guess it makes more sense to also return it.

Reply:
I should be OK now.